tests/phpunit/tests/editor/disableBlockEditorForNavigationPostType.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * @group editor
+ *
+ * @covers ::_disable_block_editor_for_navigation_post_type
+ */
+class Tests_Editor_DisableBlockEditorForNavigationPostType extends WP_UnitTestCase {
+
+	const NAVIGATION_POST_TYPE     = 'wp_navigation';
+	const NON_NAVIGATION_POST_TYPE = 'wp_non_navigation';
+
```

Comment:
In the original test, it was enabling the editor support in setUp. Is this needed in this test? Does seem to be as the test is passing. But asking to ensure.

https://github.com/WordPress/gutenberg/blob/release/13.6/phpunit/navigation-test.php#L12-L18

```php
	public function setUp() {
		$this->enable_editor_support();
	}

	public function tearDown() {
		$this->enable_editor_support();
	}
```

Reply:
I can't recall why it was implement this way in the original class.
This is not needed in `Tests_Editor_DisableContentEditorForNavigationPostType`. This class has no assertions related to block editor support for navigation posts.
But this is needed in `Tests_Editor_EnableContentEditorForNavigationPostType`.

Reply:
Looking at `_disable_block_editor_for_navigation_post_type()`, it's just a check for `'wp_navigation'` and isn't currently affecting other tests or being affected by other tests.

```php
function _disable_block_editor_for_navigation_post_type( $value, $post_type ) {
	if ( 'wp_navigation' === $post_type ) {
		return false;
	}

	return $value;
}
```

Reply:
Yes, `_disable_block_editor_for_navigation_post_type ` is kind of useless (if I may say so), but in my opinion we should still have unit tests for it since it's in Core now.
I would like to hear your suggestion. Any ideas on how to improve this unit test, @costdev ?

Reply:
- Merging the two test methods and using a data provider would mean, if more post types are included in future, more datasets can just be added to the provider.
- The assertion would change to `assertSame()` so that a `(bool) $expected` parameter can be passed.
- In addition, I would pass `(bool) $value` and `(string) $post_type` arguments to the test method.

There would be four datasets at the moment:
```php
'non-navigation post type and true' => array(
    'value'     => true,
    'post_type' => static::NON_NAVIGATION_POST_TYPE,
    'expected'  => true,
),
'non-navigation post type and false' => array(
    'value'     => false,
    'post_type' => static::NON_NAVIGATION_POST_TYPE,
    'expected'  => false,
),
'navigation post type and true' => array(
    'value'     => true,
    'post_type' => static::NAVIGATION_POST_TYPE,
    'expected'  => false,
),
'navigation post type and false' => array(
    'value'     => false,
    'post_type' => static::NAVIGATION_POST_TYPE,
    'expected'  => false,
),
```

I think that's all for the _unit_ test coverage of this function and that should cover the current implementation.

In future, we may find that integration tests could improve coverage using datasets for all registered post types to ensure there are no edge cases. This would use either `get_post_types()`, or the `$wp_post_types` global directly.

Reply:
Thank you for the feedback, @costdev.
I thought it would be enough if the test covered all lines of the method. However, I agree that the data provider is better for covering edge cases.
Fixed in 9aa06d424028af9187d41fc45b89e0b8ec2dfac2